tools/catalog-export/src/transform/toTypescript.ts

```diff
@@ -62,6 +69,37 @@ function uniqueVariable(title: string, nameScope: TakenNamesSet = GlobalNameScop
     return variableName;
 }
 
+/**
+ * This is a wrapper on top of uniqueVariable. It is useful when naming date data set properties within the DateDatasets mapping.
+ *
+ * Date data sets have a convention where date data set title is "Date (data set name)". As the context of date data sets
+ * is already established, noone cares about "Date" in the property name.
+ *
+ * This function will use just the title in parenthesis for the data set property name.
+ *
+ * @param title
+ * @param nameScope
+ */
+function dateDataSetProperty(title: string, nameScope: TakenNamesSet = GlobalNameScope): string {
+    const datasetStart = title.lastIndexOf("(");
+
+    if (datasetStart === -1) {
+        return uniqueVariable(title, nameScope);
+    }
+
+    return uniqueVariable(title.substr(datasetStart), nameScope);
```

Comment:
What about the closing paren here? I don't see it removed anywhere.

Reply:
the uniqueVariable takes care of this. its responsibility to make title into a valid and unique variable name.